### PR TITLE
ci: run k8s tests against the history

### PR DESCRIPTION
**Pull request by lebauce**
skip skydive-go-fmt
skip skydive-functional-tests-backend-elasticsearch
skip skydive-functional-tests-backend-orientdb
skip skydive-scale-tests
skip skydive-cdd-overview-tests
skip skydive-unit-tests
skip skydive-compile-tests
skip skydive-ppc64le-tests

### CODE REVIEW

**Comment by lebauce** [comment]
run skydive-scale-tests

**Comment by nplanel** [comment]
run skydive-k8s-tests

File: scripts/ci/install-istio.sh [hunk 1]
```diff
@@ -3,17 +3,17 @@
 DIR=$(dirname $0)
 
 OS=linux
-TARGET_DIR=/usr/bin
+TARGET_DIR=/opt
```
Comment (hunchback) [inline scripts/ci/install-istio.sh]: will not work as you need a path which exists both for `root` and `vagrant` users

File: scripts/ci/run-k8s-tests.sh [hunk 2]
```diff
@@ -19,7 +19,8 @@ k8s_setup
 network_setup
 WITH_EBPF=true
 WITH_K8S=true
-TEST_PATTERN=K8s
+WITH_ISTIO=true
```
Comment (hunchback) [inline scripts/ci/run-k8s-tests.sh]: I rather this be a separate test, as both k8s and istio each can exist independently (note that istio is decoupled from k8s and can be used in conjunction with docker + consul).

You also may find that k8s tests pass which when running with istio they fail ...

File: scripts/ci/run-tests-utils.sh [hunk 3]
```diff
@@ -16,12 +16,22 @@ tests_run() {
...
+        BACKEND=${BACKEND:-memory}
+        case "$BACKEND" in
+        "orientdb")
+                export ORIENTDB_ROOT_PASSWORD=root
+                ARGS="$ARGS -analyzer.topology.backend orientdb -analyzer.flow.backend orientdb"
+        ;;
+        "elasticsearch")
+                ARGS="$ARGS -analyzer.topology.backend elasticsearch -analyzer.flow.backend elasticsearch"
+        ;;
+        esac
```
Comment (hunchback) [inline scripts/ci/run-tests-utils.sh]: you can place this line here and remove lines 25, 28:

```
ARGS="$ARGS -analyzer.topology.backend $BACKEND -analyzer.flow.backend $BACKEND"
```

File: scripts/ci/install-istio.sh (hunk 1 above)
Comment (lebauce) [reply scripts/ci/install-istio.sh]: Right now, `kubectl` is called for `TestIstioDestinationRuleNode`...

File: scripts/ci/run-k8s-tests.sh (hunk 2 above)
Comment (lebauce) [reply scripts/ci/run-k8s-tests.sh]: Right now, we only support Istio with K8s. I don't want to have 2 jobs that spend 80% of their time setting up Minikube

**Comment by lebauce** [comment]
@hunchback It seems the object under `Metadata.K8s` can not be marshalled properly

**Comment by lebauce** [comment]
run skydive-k8s-tests

File: scripts/ci/install-istio.sh (hunk 1 above)
Comment (hunchback) [reply scripts/ci/install-istio.sh]: yes that is fine

**Comment by hunchback** [comment]
Have you identified what object within Metadata.K8s does not marshel? The fix would to to handle that type within SetFieldAndNormalize()

**Comment by hunchback** [comment]
I created PR https://github.com/skydive-project/skydive/pull/1381 which increases number of support fields allowing for this PR to pass. Note that you need to delete the content of ElasticSearch between every execution of the tests:

```
curl -X DELETE http://localhost:9200/_all
```

**Pull request closed without merging**